codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketServerExtensionHandler.java

```diff
@@ -62,44 +62,61 @@ public WebSocketServerExtensionHandler(WebSocketServerExtensionHandshaker... ext
         this.extensionHandshakers = Arrays.asList(extensionHandshakers);
     }
 
+    /**
+     * Constructor
+     *
+     * @param request
+     *      The instance of WebSocket upgrade HTTP request.
+     * @param extensionHandshakers
+     *      The extension handshaker in priority order. A handshaker could be repeated many times
+     *      with fallback configuration.
+     */
+    public WebSocketServerExtensionHandler(HttpRequest request,
+                                           WebSocketServerExtensionHandshaker... extensionHandshakers) {
+        this(extensionHandshakers);
+        processHttpRequest(request);
```

Comment:
I didn't look at the whole PR yet (sorry super busy) but I really dont like to call a private method from the constructor. This looks more like a code smell than a code thing to me.

Reply:
Agree. The only way to work this around I can think of is to have a static factory method, the same way it works for creating a new handshaker providing the instance of a request (the problem remains tho', just being re-shaped a bit). Maybe you can suggest another approach here from a coding perspective?